frontend/src/metabase/query_builder/components/CompareAggregations/components/OffsetPresets/OffsetPresets.tsx

```diff
@@ -0,0 +1,95 @@
+import { t } from "ttag";
+
+import { Flex, Button } from "metabase/ui";
+import * as Lib from "metabase-lib";
+import type { TemporalUnit } from "metabase-types/api";
+
+type Preset = {
+  bucket: Lib.Bucket;
+  displayName: string;
+  shortName: TemporalUnit;
+};
+
+type Props = {
+  query: Lib.Query;
+  stageIndex: number;
+  bucket: TemporalUnit | null;
+  column: Lib.ColumnMetadata;
+  onBucketChange: (bucket: TemporalUnit | null) => void;
+  onShowOffsetInput: () => void;
+};
+
+const PREFERRED_PRESETS = ["month", "year", "quarter", "week"];
+
+export function OffsetPresets({
+  query,
+  stageIndex,
+  bucket,
+  onBucketChange,
+  column,
+  onShowOffsetInput,
+}: Props) {
+  const presets = getPreferredPresets(query, stageIndex, column, bucket);
+
+  return (
+    <Flex align="flex-end" pos="relative" gap="sm">
+      {presets.map(preset => (
+        <Button
+          key={preset.shortName}
+          variant={preset.shortName === bucket ? "filled" : "default"}
+          radius="xl"
+          p="sm"
+          onClick={() => onBucketChange(preset.shortName)}
+        >
+          {preset.displayName}
+        </Button>
+      ))}
+
+      <Button variant="default" radius="xl" p="sm" onClick={onShowOffsetInput}>
+        {t`Custom...`}
+      </Button>
+    </Flex>
+  );
+}
+
+function getPreferredPresets(
```

Comment:
This looks complex and I don't fully get it. Can it be simplified? Is this the same?

```
const availabeBuckets = Lib.availableTemporalBuckets(
  query,
  stageIndex,
  column,
);

return availableBuckets.reduce((presets, bucket) => {
  const bucketInfo = Lib.displayInfo(query, stageIndex, bucket);
  if (bucketInfo.shortName === bucket || PREFERRED_PRESETS.include(bucketInfo.shortName)) {
    presets.push({ 
      bucket, 
      shortName: bucketInfo.shortName, 
      displayName: t`Previous ${info.displayName.toLowerCase()}`
    });
}, []);
```

Reply:
I've simplified the function a bit more, but it's trickier than your implementation: we need to preserve order and make sure the currently selected bucket is shown.